cockatrice/src/chatview/chatview.cpp

```diff
@@ -474,6 +490,28 @@ void ChatView::clearChat()
     lastSender = "";
 }
 
+void ChatView::redactMessages(const QString &userName, int amount)
+{
+    auto &messagePositions = userMessagePositions[userName];
+    bool removedLastMessage = false;
+    QTextCursor cursor(document());
+    for (; !messagePositions.isEmpty() && amount != 0; --amount) {
+        auto position = messagePositions.takeLast();   // go backwards from last message
```

Comment:
This is very complicated

Reply:
and? qt's cursor for document editing is trash?